### PR TITLE
Ardenian patch 1

### LINKED ISSUE #235: Add translations
https://dotd.idrinth.de/static/userscript-translation/[Language Code]/[branch]/ is now avaible,

The branch is optional and will default to master, while the language code will default to en.

### PR DESCRIPTION
**Pull request by Ardenian**
fixes #235

### CODE REVIEW

File: src/stable.js [hunk 1]
```diff
@@ -119,6 +119,7 @@ var idrinth = {
             idrinth.user.start ( );
             idrinth.names.start ( );
             idrinth.raids.start ( );
+            idrinth.text.start()
```
Comment (Idrinth) [inline src/stable.js]: where is the semicolon? :P

**Comment by Idrinth** [comment]
besides a setting to set the language code it does now work.
Resetting the default might make some sense, so we don't force english on everyone by accident.

File: src/mods/text.js [hunk 2]
```diff
@@ -42,7 +43,7 @@ idrinth.text = {
      * See languages/en.json for an example
      * @type {object}
      */
-    data: JSON.parse ( '###LANG###' ),
+    data: JSON.parse ( language+'.json' ),
```
Comment (Idrinth) [inline src/mods/text.js]: this can't have been tested, it'll break the whole thing.

File: src/mods/text.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ idrinth.text = {
             idrinth.text.initialized = true;
             return;
         }
+        idrinth.settings.change( "language", language );
```
Comment (Idrinth) [inline src/mods/text.js]: lgtm

**Review by Idrinth — CHANGES_REQUESTED** [review]

**Comment by Idrinth** [comment]
"Resetting the default might make some sense, so we don't force english on everyone by accident." was refering to our currently set default value.
If you could move the setting, so that the language is "reset"

File: src/mods/text.js [hunk 4]
```diff
@@ -10,7 +10,8 @@ idrinth.text = {
      * @returns {undefined}
      */
     start: function ( ) {
-        var language = idrinth.settings.get ( "language" ) || window.navigator.userLanguage || window.navigator.language;
+        var language = idrinth.settings.get( "lang" ) || window.navigator.userLanguage || window.navigator.language || idrinth.settings.get( "language" );
```
Comment (Idrinth) [inline src/mods/text.js]: the 4th option makes no sense

File: src/mods/settings.js [hunk 5]
```diff
@@ -21,6 +21,7 @@ idrinth.settings = {
         alarmActive: false,
         bannedRaids: { },
         language: 'en',
```
Comment (Idrinth) [inline src/mods/settings.js]: why is this still around?

**Review by Idrinth — DISMISSED** [review]

File: src/mods/text.js (hunk 4 above)
Comment (Ardenian) [reply src/mods/text.js]: I did change it: https://github.com/IDotD/Userscript/pull/237/commits/0aece27e43cbf73ffdd86f46f8308d8e51eee3e8

File: src/mods/settings.js (hunk 5 above)
Comment (Ardenian) [reply src/mods/settings.js]: Changed this one as well: https://github.com/IDotD/Userscript/pull/237/commits/8274fa6b1df818280cfa892314d862ef068b1629

Maybe something got mixed up ?

**Comment by Idrinth** [comment]
my mistake, did everything work when you tested it live?

**Comment by Ardenian** [comment]
Well, at first it broke my Raids tab: 
issue
However, I was not able to reproduce this issue after refreshing. 
Could you take a short look on testing, please ? I changed my browser's language setting, however, it does not actually change the language globally ( all sites, not only the script). Not sure what's causing that, nonetheless, English showed up in the user script as language.

**Comment by Idrinth** [comment]
>  I changed my browser's language setting, however, it does not actually change the language globally ( all sites, not only the script). Not sure what's causing that, nonetheless, English showed up in the user script as language.

The language is submitted as a request to websites, they decide what(if anything) to do with it.
The script will already have saved the "en" when you opened it for testing before changing it I assume, maybe have a look at the three relevant values in your case? :)

**Comment by Idrinth** [comment]
Generally speaking it seems to work - with a small amount of exceptions:
de

**Comment by Ardenian** [comment]
1. I missed setting.language previously, it should be fixed now with the previous commit.
2. ui.settingInfo seems to have a valid path though, it should show up. 
These two exceptions ?

The input field does not show up for me: 
settings

Has it a script dependency other than idrinth.de, like ajax.googleapi.com or something ?
I get these two error messages in my Firefox console: 
> [IDotDS] JSON.parse: unexpected end of data at line 1 column 1 of the JSON data
> "[IDotDS] Request to https://dotd.idrinth.de/armorgames/raid-service/-1/ failed."

**Comment by Idrinth** [comment]
The error sounds like my server failed to answer when you requested raids, will have a look, but it's unrelated.
You are testing your own version, right?

```
var sc = document.createElement ( 'script' );
sc.setAttribute ( 'src', 'https://dotd.idrinth.de/static/userscript/Ardenianpatch1/' );
document.getElementsByTagName ( 'body' )[0].appendChild ( sc );
```

regarding the setting.language, remember to adjust the translation files as well ;)

edit: we seem to be missing translations for the tabs as well

**Comment by Idrinth** [comment]
LGTM

**Comment by Idrinth** [comment]
Approved? :D I hate changing checks

[Approved with PullApprove](https://pullapprove.com/IDotD/Userscript/pull-request/237/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=IDotD/Userscript)

**Comment by Idrinth** [comment]
:+1:

**Comment by Idrinth** [comment]
lgtm

[Approved with PullApprove](https://pullapprove.com/IDotD/Userscript/pull-request/237/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=IDotD/Userscript)

**Pull request merged**